sre_maturity.md

```diff
@@ -0,0 +1,115 @@
+Formalise it and share it.
+
+If things go really bad, you write up an RCA.
+You should be starting to build up a library of RCAs that can feed into doing more proactive things as you mature e.g. Incident rehearsals.
+Each RCA includes a 'blameless' post mortem session with the people involved while the incident is still fresh in their head.
+This post mortem feeds important reliability improvements back to the team.
+
+#### Room to improve
+
+Your team has sufficient time in their day to pursue improvement.
+Your team needs room to practice processes, improve them and automate things.
+Without this room, your team will mature more slowly and get bogged down in just keeping the service running day to day.
+This can lead to scaling problems, both for the service and the team.
+It isn't necessary to formally measure how much time is spent on keeping things running vs. feature development or automation.
+That can come later, however you are familiar with the concept of Toil.
+
+
+## Phase 2: The ‘Walk’ Phase (WIP)
+
+### Overview
+
+Your team is well established, has working processes and is familiar with SRE principles.
+A lot of those principles are baked into their processes & daily work.
+In this phase your team is building on the foundations of observability and starting to reflect on their learnings about the service. 
+Feedback loops, automation and 'shifting left' with different types of testing are the main theme of this phase.
+Your team is capable of scaling the service and number of users in a profitable way.
+
+### Characteristics
+
+* SLOs - SLOs have been iterated on, and you have more than 1 meaningful SLO. You understand and track error budgets
+* Observability
+    * Actionable Alerts
+    * Self healing
+        * Avoid SRE having sub-systems that automagically work around the inefficiencies of upstream - needs a good argument why it can’t be fixed upstream - pushing left
+    * Combination of internal (whitebox) and external (blackbox) monitoring - tied to SLOs
+    * Dashboards, logging (easily accessible)
+* Chaos Testing (possible candidate for Phase 1, but what would that look like? Ad-hoc testing?)
+* Starting to get Exec support (improvment on 'buy-in')
+* Start measuring Toil more formally & management track it
+    * Develop relationship with the upstream (eng and sre, or sre & upstream product) for managing toil
+* Improving the architecture for lower cost to serve (Is there a prereq here - calculating & knowing the cost to serve)
+* DR & Incident Rehearsals (Identifying failure modes could be a prerequisite in phase 1)
+
+## Phase 3: The 'Run' Phase (WIP)
+
+### Overview
+
+Your team has found a balance between keeping things running and building reliability into the service.
+They have the ability to scale the service without having to increase team size.
+There is sufficient confidence in the service that your team looks at 'shifting right' with things like canary releases and production testing.
```

Comment:
I am wondering if there will be a phase 4?  Because we should capture a phase that includes proficiency at these types of things.

Reply:
If there's enough meat in an area to have a 'doing it' and a 'proficient' phase, sure e.g. chaos testing vs. chaos engineering.
Alternatively they move to an earlier phase, but that may be overloading that phase. I've done that with SLOs & considering it for Chaos testing (pending feedback above).

I don't think I have sufficient knowledge captured on canary releases & production testing to make a decision here yet.